### PR TITLE
[Optimize] Using sorted schema change processing to merge the data

### LINKED ISSUE #3660: [Alter] The rollup index's row number is same as base table.
**Describe the bug**
After creating a rollup with less key column, we found that the row number of this rollup
is just same as the base table.

**To Reproduce**
1. create a table with column k1, k2, k3, v1(sum)
2. load some data.
3. create rollup with column k1, v1(sum)

after rollup finished, the num rows of rollup is same as base table's.

**Expected behavior**
Because the number of key columns in rollup is less than base table, the data
should be more aggregate in rollup, so the row num of the rollup should be less than
base table.

### PR DESCRIPTION
**Pull request by morningman**
When reducing the key columns of an aggregate table by doing schema change
or rollup, we should trigger a re-sort schema change process to re-sort
and merge the data into new schema. So that the data in new schema will
be more aggregated.

Fix: #3660

### CODE REVIEW

File: be/src/olap/schema_change.cpp [hunk 1]
```diff
@@ -1859,6 +1859,29 @@ OLAPStatus SchemaChangeHandler::_parse_request(TabletSharedPtr base_tablet,
...
+    //      new keys:    A   X   B   C   D
+    //
+    //      old keys:    A   B   C   D
+    //      new keys:    X   A   B   C   D
+    //
+    //      old keys:    A   B   C   D
+    //      new keys:    A   B   C
+    //
+    //      old keys:    A   B   C   D
+    //      new keys:    A   B
```
Comment (chaoyli) [inline be/src/olap/schema_change.cpp:1875]: After this pull request.
old keys: A B C D
new keys: A B C
will be resorted.

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/olap/schema_change.cpp (hunk 1 above)
Comment (morningman) [reply be/src/olap/schema_change.cpp:1875]: Sorry that this PR is not ready, I will consider more situation.

Comment (EmmyMiao87) [reply be/src/olap/schema_change.cpp:1875]: This case also need be re-agg, so the sc_sorting must be marked as true.

**Comment by EmmyMiao87** [comment]
#4587

**Pull request closed without merging**